tests/test_kafka_scanner.py

```diff
@@ -378,24 +377,46 @@ def test_kafka_scan_resume_simple_partition_after_fail(self, kafka_consumer_mock
         self.assertEqual(expected_keys.difference(all_msgkeys), set())
         self.assertEqual(len(all_msgkeys), sum_msgkeys)
 
+@patch('kafka.KafkaConsumer', autospec=True)
+class KafkaScannerDeserializationTest(BaseScannerTest):
+    class MyScanner(KafkaScanner):
+        test_count = 0
+        def process_record(self, record):
+            self.test_count += 1
+            return record
+    scannerclass = MyScanner
+
+    def test_process_record_json_compressed(self, kafka_consumer_mock):
+        msgs = [('AD{:d}'.format(i), {'body':' {:d}'.format(i)}) for i in range(1000)] + \
+                [('AD{:d}'.format(i), None) for i in range(100, 200)]
+
+        samples = get_kafka_msg_samples(msgs, msgformat='json', compress=True)
+
+        scanner, _, messages = self._get_scanner_messages(samples, 3, kafka_consumer_mock,
+                               count_variations={0: 2, 1: 3, 2: 2}, msgformat='json', decompress=True)
+        msgsdict = {m['_key']: m['body'] for m in messages}
+        self.assertEqual(len(set(msgsdict)), 900)
+        self.assertEqual(scanner.scanned_count, 1100)
+        self.assertEqual(scanner.issued_count, 900)
+        self.assertEqual(scanner.deleted_count, 100)
+        self.assertEqual(scanner.dupes_count, 100)
+        self.assertEqual(scanner.test_count, 900)
+        for i in range(100, 200):
+            self.assertTrue('AD%.3d' % i not in msgsdict)
+
+    def test_process_record_json_uncompressed(self, kafka_consumer_mock):
+        msgs = [('AD{:d}'.format(i), {'body':' {:d}'.format(i)}) for i in range(1000)] + \
+                [('AD{:d}'.format(i), None) for i in range(100, 200)]
+        samples = get_kafka_msg_samples(msgs, msgformat='json', compress=False)
+        scanner, _, messages = self._get_scanner_messages(samples, 3, kafka_consumer_mock,
+                               count_variations={0: 2, 1: 3, 2: 2}, msgformat='json', decompress=False)
+        msgsdict = {m['_key']: m['body'] for m in messages}
+        self.assertEqual(len(set(msgsdict)), 900)
+        self.assertEqual(scanner.scanned_count, 1100)
+        self.assertEqual(scanner.issued_count, 900)
+        self.assertEqual(scanner.deleted_count, 100)
+        self.assertEqual(scanner.dupes_count, 100)
+        self.assertEqual(scanner.test_count, 900)
+        for i in range(100, 200):
+            self.assertTrue('AD%.3d' % i not in msgsdict)
 
-#     def test_kafka_scan_resume(self, kafka_consumer_mock, batchsize=100):
```

Comment:
please, don't remove this test.

Reply:
Actually, now that I see, [this whole code block is commented in master](https://github.com/scrapinghub/kafka-scanner/blob/master/tests/test_kafka_scanner.py#L382). Are you sure you want to keep it in master? Saving code that you might use in the future is not  good practice (YAGNI probably), and that is what git history is for.

Reply:
you are right. I hand't noticed this was commented test.